Port NXP ADSP soc/boards to HWMv2

iuliana-prodan opened the pull request:
Port NXP ADSP SoC family and boards to HWMv2.

iuliana-prodan committed:
samples: update board name

Update board name and rename files for
openamp_rsc_table sample.

Signed-off-by: Iuliana Prodan <iuliana.prodan@nxp.com>

nordicjm commented on soc/nxp/imx/Kconfig:
```diff
@@ -0,0 +1,14 @@
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_FAMILY_IMX
+	bool
```
```suggestion
```

nordicjm commented on soc/nxp/imx/imx8/Kconfig.defconfig:
```diff
@@ -0,0 +1,56 @@
+
+if SOC_SERIES_IMX_8
+
+config SOC_SERIES
+	string
+	default "imx8"
```
`Kconfig.soc`

nordicjm commented on soc/nxp/imx/imx8/Kconfig.defconfig:
```diff
@@ -0,0 +1,56 @@
+
+if SOC_MIMX8QM_ADSP
+
+config SOC
+	default "mimx8qm6"
+
+config SOC_TOOLCHAIN_NAME
+	string
+	default "nxp_imx_adsp"
```
`Kconfig.soc`

nordicjm replied:
use the `if` statement

nordicjm commented on soc/nxp/imx/imx8/Kconfig.soc:
```diff
@@ -0,0 +1,35 @@
+config SOC_PART_NUMBER_MIMX8QM6AVUFF
+	bool
+
+if SOC_MIMX8QM_ADSP
```
```suggestion
```

nordicjm commented on soc/nxp/imx/imx8/Kconfig.soc:
```diff
@@ -0,0 +1,35 @@
+	string
+	default SOC_PART_NUMBER_MIMX8QM_DSP if SOC_MIMX8QM_ADSP
+
+endif # SOC_MIMX8QM_ADSP
```
```suggestion
```

nordicjm commented on soc/nxp/imx/imx8/Kconfig.soc:
```diff
@@ -0,0 +1,35 @@
+		choice defines the default value for this string.
+
+config SOC_PART_NUMBER
+	string
```
```suggestion
```

nordicjm commented on soc/nxp/imx/imx8/Kconfig.soc:
```diff
@@ -0,0 +1,35 @@
+
+if SOC_MIMX8QM_ADSP
+
+config SOC_PART_NUMBER_MIMX8QM_DSP
+	string
+	default "MIMX8QM6AVUFF_dsp" if SOC_PART_NUMBER_MIMX8QM6AVUFF
+	help
+		This string holds the full part number of the SoC. It is a hidden
+		option that you should not set directly. The part number selection
+		choice defines the default value for this string.
+
```
```suggestion
```

nordicjm commented on soc/nxp/imx/imx8/Kconfig.soc:
```diff
@@ -0,0 +1,35 @@
+
+config SOC_PART_NUMBER
+	string
+	default SOC_PART_NUMBER_MIMX8QM_DSP if SOC_MIMX8QM_ADSP
```
```suggestion
	default "MIMX8QM6AVUFF_dsp" if SOC_PART_NUMBER_MIMX8QM6AVUFF
```

nordicjm commented on soc/nxp/imx/imx8m/CMakeLists.txt:
```diff
@@ -0,0 +1,27 @@
+
+if(CONFIG_SOC_MIMX8MP_ADSP)
+
+zephyr_include_directories(adsp)
+add_subdirectory(adsp)
+
+zephyr_include_directories(include)
+
+#  west sign
+
+# See detailed comments in soc/intel/intel_adsp/common/CMakeLists.txt
+add_custom_target(zephyr.ri ALL
+  DEPENDS ${CMAKE_BINARY_DIR}/zephyr/zephyr.ri
+)
+
+add_custom_command(
+  OUTPUT ${CMAKE_BINARY_DIR}/zephyr/zephyr.ri
+  COMMENT "west sign --if-tool-available --tool rimage ..."
+  COMMAND  west sign --if-tool-available --tool rimage --build-dir ${CMAKE_BINARY_DIR} ${WEST_SIGN_OPTS}
+  DEPENDS ${CMAKE_BINARY_DIR}/zephyr/${KERNEL_ELF_NAME}
+)
+
+set(SOC_LINKER_SCRIPT ${CMAKE_CURRENT_SOURCE_DIR}/linker.ld CACHE INTERNAL "")
```
indent with 2 spaces

nordicjm commented on soc/nxp/imx/imx8m/Kconfig.defconfig:
```diff
@@ -0,0 +1,12 @@
+
+if SOC_SERIES_IMX_8M
+
+config SOC_SERIES
+	string
+	default "imx8m"
```
`Kconfig.soc`

nordicjm commented on soc/nxp/imx/imx8m/Kconfig.defconfig.imx8mp_adsp:
```diff
@@ -0,0 +1,55 @@
+
+if SOC_MIMX8MP_ADSP
+
+config SOC
+	default "mimx8ml8"
+
+config SOC_TOOLCHAIN_NAME
+	string
+	default "nxp_imx8m_adsp"
```
`Kconfig.soc`

iuliana-prodan replied:
This SOC_TOOLCHAIN_NAME will not apply for all cores from SoC imx8m. This is only for DSP, which uses xtensa: i.e. /opt/zephyr-sdk-0.16.4/xtensa-nxp_imx8m_adsp_zephyr-elf/

nordicjm commented on soc/nxp/imx/imx8m/Kconfig.soc:
```diff
@@ -0,0 +1,39 @@
+config SOC_PART_NUMBER_MIMX8ML8CVNKZ
+	bool
+
+if SOC_MIMX8MP_ADSP
+
+config SOC_PART_NUMBER_MIMX8MP_DSP
+	string
+	default "MIMX8ML8DVNLZ_dsp" if SOC_PART_NUMBER_MIMX8ML8DVNLZ
+	default "MIMX8ML8CVNKZ_dsp" if SOC_PART_NUMBER_MIMX8ML8CVNKZ
+	help
+	  This string holds the full part number of the SoC. It is a hidden option
+	  that you should not set directly. The part number selection choice defines
+	  the default value for this string.
+
+config SOC_PART_NUMBER
+	string
+	default SOC_PART_NUMBER_MIMX8MP_DSP if SOC_MIMX8MP_ADSP
+
+endif # SOC_MIMX8MP_ADSP
```
as above and apply the above raised issues to all other files

iuliana-prodan commented on boards/nxp/mimx8mp_evk/board.yml:
```diff
@@ -0,0 +1,8 @@
+    - name: imx8mp
+      variants:
+        - name: adsp
+          cpucluster: ''
```
see 
```
    socs:
    - name: imx8mp
      cpuclusters:
      - name: adsp
      - name: a53
      - name: m7
 ```
 and the adsp is a core on 8mp board, but doesn't have a variant, like a53: a53 or a53/smp.
 If I don't add a variant is crashes saying that the variant is missing.

nordicjm replied:
then you have done something wrong

iuliana-prodan replied:
I found the issue.
I should let in `board.yml` only
```
board:
  name: mimx8qm_mek
  vendor: nxp
  socs:
    - name: mimx8qm6
```
and the variants are taken from each yaml - like from `mimx8qm_mek_mimx8qm6_adsp.yaml`

iuliana-prodan committed:
modules: mcux: remove SOC_FAMILY_NXP_ADSP

Remove SOC_FAMILY_NXP_ADSP since this is not used
anymore for the ADSP.
With HWMv2 ADSP is part of SOC_FAMILY_IMX.

Signed-off-by: Iuliana Prodan <iuliana.prodan@nxp.com>

iuliana-prodan commented:
v2:
- fix soc name in Kconfig - not it matches the one from soc.yml;
- fix comments from @nordicjm;
- fix compliance errors.

Regarding compliance checks - I still have 2 errors.

One is regarding `nxp_adsp_rt595` - this is in process of porting to hwmv2 and it has nothing to do with this PR.

Next is:
```
Missing SoC names or CONFIG_SOC vs soc.yml out of sync:

soc name: mimx8qm6 not found in CONFIG_SOC defaults.
soc name: imx8ulp not found in CONFIG_SOC defaults.
soc name: mimx8ml8 not found in CONFIG_SOC defaults.
soc name: mimx8qx6 not found in CONFIG_SOC defaults.
```
This is false IMO. I have the  `CONFIG SOC` defined for all 4 socs.
See, for example for `mimx8qm6` - 
soc.yml - https://github.com/zephyrproject-rtos/zephyr/pull/69398/files#diff-57bf2b8b4543a2b08e7ae3d16f19acbfc8cfedd275cfd3756b1c1000f574268fR6
defconfig - https://github.com/zephyrproject-rtos/zephyr/pull/69398/files#diff-76a8a7d05373ebb56c744e770e0ec4fa8dbcde98057a5ed3a562cd46c197303aR9
So, is something wrong here?

iuliana-prodan commented:
> This PR has a dependency on #69399

The dependency on  #69399 was fixed by having the same soc name in `Kconfig` and `soc.yml`.
Now there's a dependency on https://github.com/zephyrproject-rtos/hal_nxp/pull/341

JiafeiPan commented on soc/nxp/imx/imx8m/linker.ld:
this file is also only for adsp, right, move it to directory "adsp"?

iuliana-prodan replied:
done.

I've also moved the pinctrl_soc in adsp and the one from a53 in the a53 folder

JiafeiPan commented on soc/nxp/imx/imx8m/include/memory.h:
only used by adsp and move it to "adsp" directory?

iuliana-prodan replied:
done

JiafeiPan commented on soc/nxp/imx/imx8m/include/_soc_inthandlers.h:
only used by adsp and move it to "adsp" directory?

iuliana-prodan replied:
done

nordicjm commented on soc/nxp/imx/Kconfig.soc:
```diff
@@ -0,0 +1,10 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_FAMILY_IMX
+	bool
+
+config SOC_FAMILY
+	default "nxp_imx" if SOC_FAMILY_IMX
```
```suggestion
config SOC_FAMILY_NXP_IMX
	bool

config SOC_FAMILY
	default "nxp_imx" if SOC_FAMILY_NXP_IMX
```

JiafeiPan replied:
I suggest to use SOC_FAMILY_IMX because there are some existing device drivers are using this configuration item. so need to modify soc.yml and :

config SOC_FAMILY
	default "imx" if SOC_FAMILY_IMX

nordicjm replied:
Or update the drivers as has been done for other vendors, should be a simple `sed` command

iuliana-prodan replied:
done - s/SOC_FAMILY_IMX/SOC_FAMILY_NXP_IMX

nordicjm commented on soc/nxp/imx/imx8/CMakeLists.txt:
```diff
@@ -0,0 +1,27 @@
+# SPDX-License-Identifier: Apache-2.0
+
+if(CONFIG_SOC_MIMX8QM_ADSP)
+
```
```suggestion
```

nordicjm commented on soc/nxp/imx/imx8/CMakeLists.txt:
```diff
@@ -0,0 +1,27 @@
+  )
+
+  set(SOC_LINKER_SCRIPT ${CMAKE_CURRENT_SOURCE_DIR}/linker.ld CACHE INTERNAL "")
+
```
```suggestion
```

nordicjm commented on soc/nxp/imx/imx8/Kconfig.soc:
```diff
@@ -0,0 +1,30 @@
+	bool
+	select SOC_FAMILY_IMX
+
+config SOC_SERIES
+	string
+	default "imx8"
+
```
`if ...`

nordicjm commented on soc/nxp/imx/imx8/Kconfig.soc:
```diff
@@ -0,0 +1,30 @@
+config SOC_PART_NUMBER
+	default "MIMX8QM6AVUFF" if SOC_PART_NUMBER_MIMX8QM6AVUFF
+
+config MCUX_CORE_SUFFIX
+	string
+	default "_dsp" if SOC_MIMX8QM_ADSP
```
`Kconfig`

nordicjm commented on soc/nxp/imx/imx8m/CMakeLists.txt:
```diff
@@ -0,0 +1,27 @@
+# SPDX-License-Identifier: Apache-2.0
+
+if(CONFIG_SOC_MIMX8MP_ADSP)
+
```
```suggestion
```

nordicjm commented on soc/nxp/imx/imx8m/CMakeLists.txt:
```diff
@@ -0,0 +1,27 @@
+  )
+
+  set(SOC_LINKER_SCRIPT ${CMAKE_CURRENT_SOURCE_DIR}/linker.ld CACHE INTERNAL "")
+
```
```suggestion
```

nordicjm commented on soc/nxp/imx/imx8m/Kconfig.defconfig.mimx8ml8_adsp:
```diff
@@ -0,0 +1,55 @@
+
+if SOC_MIMX8MP_ADSP
+
+config SOC
+	default "mimx8ml8"
+
+config SOC_TOOLCHAIN_NAME
+	string
+	default "nxp_imx8m_adsp"
```
as above

iuliana-prodan replied:
done

nordicjm commented on soc/nxp/imx/imx8m/Kconfig.soc:
```diff
@@ -0,0 +1,34 @@
+	bool
+	select SOC_FAMILY_IMX
+
+config SOC_SERIES
+	string
+	default "imx8m"
```
as above

nordicjm commented on soc/nxp/imx/imx8m/Kconfig.soc:
```diff
@@ -0,0 +1,34 @@
+	default "MIMX8ML8DVNLZ" if SOC_PART_NUMBER_MIMX8ML8DVNLZ
+	default "MIMX8ML8CVNKZ" if SOC_PART_NUMBER_MIMX8ML8CVNKZ
+
+config MCUX_CORE_SUFFIX
+	string
+	default "_dsp" if SOC_MIMX8MP_ADSP
```
as above

nordicjm commented on soc/nxp/imx/imx8ulp/CMakeLists.txt:
```diff
@@ -0,0 +1,27 @@
+# SPDX-License-Identifier: Apache-2.0
+
+if(CONFIG_SOC_MIMX8ULP_ADSP)
+
```
```suggestion
```

nordicjm commented on soc/nxp/imx/imx8ulp/CMakeLists.txt:
```diff
@@ -0,0 +1,27 @@
+  )
+
+  set(SOC_LINKER_SCRIPT ${CMAKE_CURRENT_SOURCE_DIR}/linker.ld CACHE INTERNAL "")
+
```
```suggestion
```

nordicjm commented on soc/nxp/imx/imx8ulp/Kconfig.defconfig:
```diff
@@ -0,0 +1,52 @@
+
+if SOC_MIMX8ULP_ADSP
+
+config SOC
+	default "imx8ulp"
+
+config SOC_TOOLCHAIN_NAME
+	string
+	default "nxp_imx8ulp_adsp"
```
as above

nordicjm commented on soc/nxp/imx/imx8ulp/Kconfig.soc:
```diff
@@ -0,0 +1,20 @@
+	bool
+	select SOC_FAMILY_IMX
+
+config SOC_SERIES
+	string
+	default "imx8ulp"
```
as above

nordicjm commented on soc/nxp/imx/imx8x/CMakeLists.txt:
```diff
@@ -0,0 +1,27 @@
+# SPDX-License-Identifier: Apache-2.0
+
+if(CONFIG_SOC_MIMX8QXP_ADSP)
+
```
```suggestion
```

nordicjm commented on soc/nxp/imx/imx8x/CMakeLists.txt:
```diff
@@ -0,0 +1,27 @@
+  )
+
+  set(SOC_LINKER_SCRIPT ${CMAKE_CURRENT_SOURCE_DIR}/linker.ld CACHE INTERNAL "")
+
```
```suggestion
```

nordicjm commented on soc/nxp/imx/imx8x/Kconfig.defconfig:
```diff
@@ -0,0 +1,52 @@
+
+if SOC_MIMX8QXP_ADSP
+
+config SOC
+	default "mimx8qx6"
+
+config SOC_TOOLCHAIN_NAME
+	string
+	default "nxp_imx_adsp"
```
as above

nordicjm commented on soc/nxp/imx/imx8x/Kconfig.soc:
```diff
@@ -0,0 +1,34 @@
+	bool
+	select SOC_FAMILY_IMX
+
+config SOC_SERIES
```
etc.

nordicjm commented on boards/nxp/mimx8mp_evk/Kconfig.mimx8mp_evk:
```diff
@@ -0,0 +1,9 @@
+
+config BOARD_MIMX8MP_EVK
+	select SOC_MIMX8MP_ADSP if BOARD_MIMX8MP_EVK_MIMX8ML8_ADSP
+	select SOC_PART_NUMBER_MIMX8ML8DVNLZ
+	select SOC_PART_NUMBER_MIMX8ML8CVNKZ
```
how can this have 2 part numbers?

iuliana-prodan replied:
Well.. good question - I don't know :(
This is how is defined in hal - https://github.com/zephyrproject-rtos/hal_nxp/blob/master/mcux/mcux-sdk/devices/MIMX8ML8/Kconfig#L22 and I enabled the DSP from those parts.

It's ok thata re only 2 - we could have more - https://www.nxp.com/products/processors-and-microcontrollers/arm-processors/i-mx-applications-processors/i-mx-8-applications-processors/i-mx-8m-plus-arm-cortex-a53-machine-learning-vision-multimedia-and-industrial-iot:IMX8MPLUS:

iuliana-prodan replied:
also see here: https://raw.githubusercontent.com/zephyrproject-rtos/hal_nxp/master/mcux/mcux-sdk/devices/MIMX8ML8/MIMX8ML8_dsp.h
```
/*
** ###################################################################
**     Processors:          MIMX8ML8CVNKZ_dsp
**                          MIMX8ML8DVNLZ_dsp
**
**     Compiler:            XCC Compiler
**     Reference manual:    IMX8MPRM, Rev.D, 12/2020
**     Version:             rev. 5.0, 2021-03-01
**     Build:               b230302
**
```

nordicjm replied:
this cannot have 2 part numbers, select the correct part number

dbaluta replied:
@nordicjm I think it relates to how the chip is certified. @iuliana-prodan we can go with just the first part number in the list for now.

@nordicjm Is there a limitation so that we cannot support two part numbers in the same time?

nordicjm commented on boards/nxp/mimx8qm_mek/board.yml:
```diff
@@ -0,0 +1,8 @@
+  vendor: nxp
+  socs:
+    - name: mimx8qm6
+      variants:
+        - name: adsp
```
no

iuliana-prodan replied:
fix it

nordicjm commented on boards/nxp/mimx8qxp_mek/Kconfig.mimx8qxp_mek:
```diff
@@ -0,0 +1,9 @@
+
+config BOARD_MIMX8QXP_MEK
+	select SOC_MIMX8QXP_ADSP if BOARD_MIMX8QXP_MEK_MIMX8QX6_ADSP
+	select SOC_PART_NUMBER_MIMX8QX6AVLFZ
+	select SOC_PART_NUMBER_MIMX8QX6CVLDZ
```
as above

nordicjm commented on boards/nxp/mimx8qxp_mek/board.yml:
```diff
@@ -0,0 +1,8 @@
+  socs:
+    - name: mimx8qx6
+      variants:
+        - name: adsp
```
as above

nordicjm commented on boards/nxp/mimx8ulp_evk/board.yml:
```diff
@@ -0,0 +1,8 @@
+  socs:
+    - name: imx8ulp
+      variants:
+        - name: adsp
+          cpucluster: ''
```
as above

iuliana-prodan committed:
modules: rename SOC_FAMILY_IMX

Rename SOC_FAMILY_IMX to SOC_FAMILY_NXP_IMX.

I've also kept SOC_FAMILY_IMX since there are
still legacy socs/boards that use it. When all
are ported to HWMv2 this should be removed.

Signed-off-by: Iuliana Prodan <iuliana.prodan@nxp.com>

iuliana-prodan commented:
> v2:
> 
> * fix soc name in Kconfig - not it matches the one from soc.yml;
> * fix comments from @nordicjm;
> * fix compliance errors.
> 
> Regarding compliance checks - I still have 2 errors.
> 
> One is regarding `nxp_adsp_rt595` - this is in process of porting to hwmv2 and it has nothing to do with this PR.
> 
> Next is:
> 
> ```
> Missing SoC names or CONFIG_SOC vs soc.yml out of sync:
> 
> soc name: mimx8qm6 not found in CONFIG_SOC defaults.
> soc name: imx8ulp not found in CONFIG_SOC defaults.
> soc name: mimx8ml8 not found in CONFIG_SOC defaults.
> soc name: mimx8qx6 not found in CONFIG_SOC defaults.
> ```
> 
> This is false IMO. I have the `CONFIG SOC` defined for all 4 socs. See, for example for `mimx8qm6` - soc.yml - https://github.com/zephyrproject-rtos/zephyr/pull/69398/files#diff-57bf2b8b4543a2b08e7ae3d16f19acbfc8cfedd275cfd3756b1c1000f574268fR6 defconfig - https://github.com/zephyrproject-rtos/zephyr/pull/69398/files#diff-76a8a7d05373ebb56c744e770e0ec4fa8dbcde98057a5ed3a562cd46c197303aR9 So, is something wrong here?

v3: 
- addressed all comments from @nordicjm (thank you for reviewing this!)

iuliana-prodan commented:
> > This PR has a dependency on #69399
> 
> The dependency on #69399 was fixed by having the same soc name in `Kconfig` and `soc.yml`. Now there's a dependency on [zephyrproject-rtos/hal_nxp#341](https://github.com/zephyrproject-rtos/hal_nxp/pull/341)

+ a dependency on https://github.com/zephyrproject-rtos/hal_nxp/pull/342

nordicjm commented on soc/nxp/imx/imx8/Kconfig.defconfig:
```diff
@@ -1,11 +1,24 @@
-source "soc/soc_legacy/xtensa/nxp_adsp/*/Kconfig.defconfig.series"
+if SOC_MIMX8QM_ADSP
+
+config SOC
+	default "mimx8qm6"
```
Kconfig.soc

nordicjm commented on soc/nxp/imx/imx8/Kconfig.soc:
```diff
@@ -0,0 +1,30 @@
+
+config SOC_SERIES
+	string
+	default "imx8" if SOC_SERIES_IMX_8
```
Align on value and have the same for both e.g `imx8` and `SOC_SERIES_IMX8`

iuliana-prodan replied:
I've updated all to: SOC_SERIES_IMX8, SOC_SERIES_IMX8M, SOC_SERIES_IMX8X, SOC_SERIES_IMX8ULP

nordicjm commented on soc/nxp/imx/imx8m/Kconfig:
```diff
@@ -0,0 +1,19 @@
+	select PINCTRL_IMX if HAS_MCUX_IOMUXC
+
+config MCUX_CORE_SUFFIX
+	string
```
```suggestion
```

nordicjm commented on soc/nxp/imx/imx8m/Kconfig.defconfig.mimx8ml8_adsp:
```diff
@@ -0,0 +1,51 @@
+
+if SOC_MIMX8MP_ADSP
+
+config SOC
+	default "mimx8ml8"
```
as above

iuliana-prodan replied:
moved to `Kconfig.soc`

nordicjm commented on soc/nxp/imx/imx8m/Kconfig.soc:
```diff
@@ -0,0 +1,34 @@
+config SOC_SERIES_IMX_8M
+	bool
+	select SOC_FAMILY_NXP_IMX
+
+config SOC_SERIES
+	string
+	default "imx8m" if SOC_SERIES_IMX_8M
```
as above

iuliana-prodan replied:
done - removed `string` and updated the name to `SOC_SERIES_IMX8M`

nordicjm commented on soc/nxp/imx/imx8ulp/Kconfig.defconfig:
```diff
@@ -0,0 +1,48 @@
+
+if SOC_MIMX8ULP_ADSP
+
+config SOC
+	default "imx8ulp"
```
as above

iuliana-prodan replied:
moved to `Kconfig.soc`

nordicjm commented on soc/nxp/imx/imx8ulp/Kconfig.soc:
```diff
@@ -0,0 +1,24 @@
+	bool
+	select SOC_FAMILY_NXP_IMX
+
+config SOC_SERIES
+	string
+	default "imx8ulp" if SOC_SERIES_IMX_8ULP
```
as above

nordicjm commented on soc/nxp/imx/imx8x/Kconfig:
```diff
@@ -1,15 +1,16 @@
+	select HAS_MCUX
+
+config MCUX_CORE_SUFFIX
+	string
```
```suggestion
```

nordicjm commented on soc/nxp/imx/imx8x/Kconfig.defconfig:
```diff
@@ -0,0 +1,48 @@
+
+if SOC_MIMX8QXP_ADSP
+
+config SOC
+	default "mimx8qx6"
```
as above

nordicjm commented on soc/nxp/imx/imx8x/Kconfig.soc:
```diff
@@ -0,0 +1,33 @@
+	select SOC_FAMILY_NXP_IMX
+
+config SOC_SERIES
+	string
+	default "imx8x" if SOC_SERIES_IMX_8X
```
as above

nordicjm commented:
Also:

- Compliance issues need fixing
- Failing test issues need fixing

iuliana-prodan commented on soc/nxp/imx/imx8/Kconfig:
```diff
@@ -1,15 +1,16 @@
-	  Enable support for NXP i.MX8M Audio DSP
+	select HAS_MCUX
+
+config MCUX_CORE_SUFFIX
+	string
+	default "_dsp" if SOC_MIMX8QM_ADSP
```
I've moved it here, as you suggested - see https://github.com/zephyrproject-rtos/zephyr/pull/69398#discussion_r1502154676
Where is best to be added?
Of course, I'll remove also the `string`.

nordicjm replied:
`Kconfig` is fine

iuliana-prodan committed:
soc: nxp: Port i.MX ADSP family to hardware model v2

Port i.MX ADSP family to HWMv2, including series:
- imx8
- imx8m
- imx8x
- imx8ulp

Signed-off-by: Iuliana Prodan <iuliana.prodan@nxp.com>

iuliana-prodan committed:
boards: nxp: Convert i.MX ADSP boards to hardware model v2

Convert i.MX ADSP boards to HMV2 including:
- imx8mp_evk
- imx8qxp_mek
- imx8qm_mek
- imx8ulp_evk

Signed-off-by: Iuliana Prodan <iuliana.prodan@nxp.com>

iuliana-prodan committed:
west: update xtensa hal

Update Xtensa HAL to get the
rename of the SoC overlay for
NXP i.MX8ULP.

Signed-off-by: Iuliana Prodan <iuliana.prodan@nxp.com>

iuliana-prodan commented:
@nordicjm  for compliance check I get:
```
Error: See https://docs.zephyrproject.org/latest/build/kconfig/tips.html for more details.

 Found references to undefined Kconfig symbols. If any of these are false
positives, then add them to UNDEF_KCONFIG_WHITELIST in /home/runner/work/zephyr/zephyr/./scripts/ci/check_compliance.py.

If the reference is for a comment like /* CONFIG_FOO_* */ (or
/* CONFIG_FOO_*_... */), then please use exactly that form (with the '*'). The
CI check knows not to flag it.

More generally, a reference followed by $, @, {, *, or ## will never be
flagged.

CONFIG_SOC_NXP_RT595                       boards/boards_legacy/xtensa/nxp_adsp_rt595/nxp_adsp_rt595_defconfig:2
CONFIG_SOC_SERIES_NXP_RT5XX                boards/boards_legacy/xtensa/nxp_adsp_rt595/nxp_adsp_rt595_defconfig:1
```

and this part if handled in another PR by @danieldegrasse  - https://github.com/zephyrproject-rtos/zephyr/pull/69398/commits/56551e3be39b4a6459c91bd9a957485cf86509b5

nordicjm commented:
Did you rebase? There should be no compliance failures

iuliana-prodan committed:
manifest: update SOF sha for NXP HWMv2

Update SOF for NXP's new boards name and configs.

Signed-off-by: Iuliana Prodan <iuliana.prodan@nxp.com>

iuliana-prodan commented:
> Did you rebase? There should be no compliance failures

I did, multiple times, but those errors are still there. :(
Am I missing something?

iuliana-prodan commented:
@nordicjm & all, be aware that some `intel_adsp` tests will fail because I've update the sof repo [here](https://github.com/zephyrproject-rtos/zephyr/pull/69398/commits/0517439ec6dc3f0618f7fcb7c0a1f1498e19d10c).
Before my change it was at [pull/39/head](https://github.com/zephyrproject-rtos/sof/pull/39) (Intel's updates for HWMv2), now I've updated to [pull/40/head](https://github.com/zephyrproject-rtos/sof/pull/40)  (NXP's updates for HWMv2).
None of these PRs is merged in [sof](https://github.com/zephyrproject-rtos/sof/pulls), yet.

nordicjm commented:
> @nordicjm & all, be aware that some `intel_adsp` tests will fail because I've update the sof repo [here](https://github.com/zephyrproject-rtos/zephyr/pull/69398/commits/0517439ec6dc3f0618f7fcb7c0a1f1498e19d10c). Before my change it was at [pull/39/head](https://github.com/zephyrproject-rtos/sof/pull/39) (Intel's updates for HWMv2), now I've updated to [pull/40/head](https://github.com/zephyrproject-rtos/sof/pull/40) (NXP's updates for HWMv2). None of these PRs is merged in [sof](https://github.com/zephyrproject-rtos/sof/pulls), yet.

You need to make a new commit on top of theirs or make a new PR with their commit included, we will not be merging any PRs where there are CI failures

nordicjm commented:
> > Did you rebase? There should be no compliance failures
> 
> I did, multiple times, but those errors are still there. :( Am I missing something?

Then the issues are with your additions here

iuliana-prodan commented:
@nordicjm all issues solved, CI is passing.
